python/ray/data/_internal/datasource/clickhouse_datasource.py

```diff
@@ -0,0 +1,201 @@
+            columns: Optional list of columns to select from the data source.
+                If no columns are specified, all columns will be selected by default.
+            filters: Optional fields and values mapping to use to filter the data via
+                WHERE clause. The value should be a tuple where the first element is
+                one of ('is', 'not', 'less', 'greater') and the second
+                element is the value to filter by. The default operator
+                is 'is'. Only strings, ints, floats, booleans,
+                and None are allowed as values.
+            order_by: Optional tuple containing a list of columns to order by and a
+                boolean indicating whether the order should be descending
+                (True for DESC, False for ASC).
+            client_settings: ClickHouse server settings to be used with the
+                session/every request
+            client_kwargs: Optional keyword arguments to pass to the
+                ClickHouse client.
+        """
+        self._entity = entity
+        self._dsn = dsn
+        self._columns = columns
+        self._filters = filters
+        self._order_by = order_by
+        self._client_settings = client_settings if client_settings is not None else {}
+        self._client_kwargs = client_kwargs if client_kwargs is not None else {}
+        self._client = None
+        self._query = self._generate_query()
+
+    def _get_or_create_client(self):
+        _check_import(self, module="clickhouse_connect", package="clickhouse-connect")
+        import clickhouse_connect
+
+        if self._client is None:
+            self._client = clickhouse_connect.get_client(
+                dsn=self._dsn,
+                settings=self._client_settings,
+                **self._client_kwargs,
+            )
+
+    def _generate_query(self) -> str:
+        select_clause = ", ".join(self._columns) if self._columns else "*"
+        query = f"SELECT {select_clause} FROM {self._entity}"
+
+        if self._filters:
+
+            def validate_non_numeric_ops(column: str, op: str) -> str:
+                non_numeric_ops = ["is", "not"]
+                if op not in non_numeric_ops:
+                    logger.warning(
+                        f"Unsupported operator '{op}' for filter on '{column}'. "
+                        f"Defaulting to 'is'"
+                    )
+                    op = "is"
+                return op
+
+            ops = {"is": "=", "not": "!=", "less": "<", "greater": ">"}
+            filter_conditions = []
+            for key, (operator, value) in self._filters.items():
+                operator = operator.lower()
+                if operator not in ops:
+                    logger.warning(
+                        f"Unsupported operator '{operator}' for filter on '{key}' "
+                        f"Defaulting to 'is'"
+                    )
+                    operator = "is"
```

Comment:
I think this fallback doesn't really make sense as it potentially drastically deviates from user intent. 

Let's just fail outright if we see unsupported op

Reply:
I attempted to resolve this using a ValueError in my latest commit.